src/main/java/carracing/domain/service/RandomFourStepRacingService.java

```diff
@@ -1,60 +1,63 @@
 package carracing.domain.service;
 
+import carracing.domain.dto.RacingData;
 import carracing.domain.dto.RacingResult;
+import carracing.domain.dto.RoundData;
 import carracing.domain.dto.RoundResult;
 import carracing.domain.entity.Car;
 import carracing.domain.entity.Challengers;
-import carracing.domain.entity.Number;
+import carracing.domain.entity.Participant;
 import carracing.domain.entity.Round;
+import carracing.domain.entity.Winner;
 import carracing.domain.utils.RandomUtils;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.function.Supplier;
+import java.util.function.BooleanSupplier;
+
+import static java.util.stream.Collectors.toList;
 
 public class RandomFourStepRacingService implements CarRacingService{
 
   private static final int STEP_FORWARD_STANDARD = 4;
-
-  private Challengers challengers;
-  private Round round;
-
-  public RandomFourStepRacingService(Number number, Round round) {
-    if (!isValidInput(number, round)) {
-      throw new IllegalArgumentException();
-    }
-    this.challengers = new Challengers(number);
-    this.round = round;
-    initRacing(number, round);
-  }
-
-  private boolean isValidInput(Number number, Round round) {
-    return number != null && round != null;
-  }
+  private static final String GAME_START_ERROR_MESSAGE = "게임을 시작할 수 없습니다.";
 
   @Override
-  public void initRacing(Number number, Round round) {
-    for (int i = 0; i < number.getValue(); i++) {
-      challengers.register(new Car());
+  public Challengers registerRacer(Participant participant) {
+    Challengers challengers = new Challengers(participant);
+    for (int id = 0; id < participant.size(); id++) {
+      challengers.register(new Car(participant.getParticipant().get(id)));
     }
+    return challengers;
   }
 
   @Override
-  public RacingResult gameStart() {
-    List<RoundResult> roundResultList = new ArrayList<>();
+  public RacingResult gameStart(Round round, Challengers challengers) {
+    validateInputValue(round, challengers);
+    List<RoundData> roundDataList = new ArrayList<>();
     while (round.hasMoreChance()) {
```

Comment:
변수를 선언하는 영역과 로직을 수행하는 부분등 개념적 유사성에 따라 수직 거리에 공백을 두면 가독성을 높힐 수 있습니다. 

```suggestion
    validateInputValue(round, challengers);
    
    List<RoundData> roundDataList = new ArrayList<>();
    
    while (round.hasMoreChance()) {
```